resemble.js

```diff
@@ -148,22 +148,40 @@ var isNode = new Function("try {return this===global;}catch(e){return false;}");
         }
 
         function withinComparedArea(x, y, width, height) {
-            var isIncluded = true;
-
-            if (
-                boundingBox !== undefined &&
-                !withinBoundingBox(x, y, width, height, boundingBox)
-            ) {
-                isIncluded = false;
+            var isIncluded = true, i, boundingBox, ignoredBox, selected, ignored;
+            if(boundingBoxs instanceof Array) {
+            	selected = false;
+            	for (i = 0; i < boundingBoxs.length; i++) {
+            		boundingBox = boundingBoxs[i];
+            		if (withinBoundingBox(x, y, width, height, boundingBox)) {
+            			selected = true;
+            			break;
+            		}
+            	}
             }
-
-            if (
-                ignoredBox !== undefined &&
-                withinBoundingBox(x, y, width, height, ignoredBox)
-            ) {
-                isIncluded = false;
+            if(ignoredBoxs instanceof Array) {
+            	ignored = true;
+            	for (i = 0; i < ignoredBoxs.length; i++) {
+            		ignoredBox = ignoredBoxs[i];
+            		if (withinBoundingBox(x, y, width, height, ignoredBox)) {
+            			ignored = false;
+            			break;
+            		}
+            	}
             }
-
+            
+            if(selected === undefined && ignored === undefined) {
+            	return true;
+            }
+            if(selected === false && ignored === true) {
```

Comment:
Could we remove these last three if statements and simply `return selected && ignored`